platforms/core-runtime/launcher/src/main/java/org/gradle/tooling/internal/provider/ProviderConnection.java

```diff
@@ -152,40 +151,19 @@ public Object run(String modelName, BuildCancellationToken cancellationToken, Pr
             throw new IllegalArgumentException("No model type or tasks specified.");
         }
         Parameters params = initParams(providerParameters);
-        if (BuildEnvironment.class.getName().equals(modelName)) {
-            //we don't really need to launch the daemon to acquire information needed for BuildEnvironment
+        if (GradleEnvironment.class.getName().equals(modelName)) {
```

Comment:
The Tooling API should declare `GradleEnvironment` as a supported model. Javadoc on `ProjectConnection.getModel()` has a list of supported models; this type should be listed there.

Reply:
@big-guy On the consumer-side, we could provide backward-compatility for  `GradleEnvironment` queries by requesting the `BuildEnvironment` model for old versions.